allennlp/training/metrics/auc.py

```diff
@@ -82,7 +87,30 @@ def __call__(
             [self._all_gold_labels, torch.masked_select(gold_labels, mask).long()], dim=0
         )
 
+        if is_distributed():
+            try:
+                # The following logic will only work if the batches are of equal length.
```

Comment:
Is that a safe assumption?

Reply:
Batches will be of equal length in all cases except when using `max_tokens_batch_sampler`. Currently, we emit a warning in that case. Can change it to error.

Reply:
I'd prefer an error. Downstream components may make choices based on the metrics (such as early stopping), and a warning after 30 minutes of training is easy to miss.

Reply:
Oh, I see you've already done this. Just not in this here diff :-)